src/System.Runtime/tests/Performance/Perf.String.cs

```diff
@@ -232,28 +233,104 @@ public void Substring_int_int(int size)
                         testString.Substring(startIndex, length);
         }
 
-        [Benchmark]
-        [MemberData(nameof(TestStringSizes))]
-        public void ToLower(int size)
+        [Flags]
+        public enum ChangeCaseOptions
         {
-            PerfUtils utils = new PerfUtils();
-            string testString = utils.CreateString(size);
+            None = 0,
+            UniqueString = 0x1,
+
+            MiddleTurkishI = 0x2,
+            MiddleDifferentCase = 0x4,
+            AllDifferentCase = 0x8,
+        }
+
+        public static IEnumerable<object[]> ChangeCaseMemberData() =>
+            from size in new[] { 1, 10, 500 }
+            from culture in new[] { "en-US" }
+            from options in new[]
+            {
+                ChangeCaseOptions.None,
+                ChangeCaseOptions.MiddleTurkishI,
+                ChangeCaseOptions.MiddleDifferentCase,
+                ChangeCaseOptions.AllDifferentCase,
+
+                ChangeCaseOptions.UniqueString,
+                ChangeCaseOptions.UniqueString | ChangeCaseOptions.MiddleTurkishI,
+                ChangeCaseOptions.UniqueString | ChangeCaseOptions.MiddleDifferentCase,
+                ChangeCaseOptions.UniqueString | ChangeCaseOptions.AllDifferentCase,
+            }
+            select new object[] { size, options, culture };
+
+        private static string CreateChangeCaseString(int size, ChangeCaseOptions options, bool upper)
+        {
+            Span<char> chars = new char[size];
+
+            bool differentCase = (options & ChangeCaseOptions.AllDifferentCase) != 0;
+            chars.Fill(upper != differentCase ? 'S' : 's');
+
+            if ((options & ChangeCaseOptions.MiddleTurkishI) != 0)
```

Comment:
I don't have the impl in front of me but it cuts over to a slow path when it hits a non ASCII character right? Why cut over in the middle? If you're testing perf perhaps you want to have entirely slow case and entirely fast case to not mix what you test?

Reply:
I wanted the perf test to assume as little as possible about the implementation, e.g. whether it goes front to back or back to front.  Figured I'd just change the middle then.

Don't have a strong opinion, though.

Reply:
Unless you expect us to change the implementation, I think it's fine for the perf test to attempt to highlight different aspects. But I don't care either.

Reply:
Ok.  I'm going to leave it for now.  The other reason for this is that I'm already bumping up against how many tests seem reasonable for this (i.e. don't take too much time), and if we wanted to have the extremes, we'd want tests for when the non-ASCII char occurs at the beginning and at the end.

As an aside, it'd be great if the perf test harness supported a filter like `/p:XunitOptions="-method ..."`.  That appears to be completely ignored, and I don't see any other way (other than deleting all the other tests in the project temporarily) to only run a subset of tests.